projects/plugins/jetpack-mu-wpcom/src/features/coming-soon/coming-soon.php

```diff
@@ -0,0 +1,235 @@
+	add_filter( 'jetpack_implode_frontend_css', '__return_false', 99 ); // Jetpack "implodes" all registered CSS files into one file.
+	add_filter( 'woocommerce_demo_store', '__return_false' ); // Prevent the the wocommerce demo store notice from displaying.
+	add_filter( 'jetpack_open_graph_image_default', 'A8C\FSE\Coming_soon\coming_soon_share_image' ); // Set the default OG image.
+	add_filter( 'jetpack_twitter_cards_image_default', 'A8C\FSE\Coming_soon\coming_soon_share_image' ); // Set the default Twitter Card image.
+	wp_enqueue_style( 'recoleta-font', '//s1.wp.com/i/fonts/recoleta/css/400.min.css', array(), A8C_ETK_PLUGIN_VERSION );
+
+	include __DIR__ . '/fallback-coming-soon-page.php';
+}
+
+/**
+ * This filter makes sure it's possible to fetch `wpcom_public_coming_soon` option via public-api.wordpress.com.
+ *
+ * @param object $options array Retrieved site options, ready to be returned as API respomnse.
+ *
+ * @return array current value of `wpcom_public_coming_soon`
+ */
+function add_public_coming_soon_to_settings_endpoint_get( $options ) {
+	$wpcom_public_coming_soon            = (int) get_option( 'wpcom_public_coming_soon' );
+	$options['wpcom_public_coming_soon'] = $wpcom_public_coming_soon;
+
+	return $options;
+}
+add_filter( 'site_settings_endpoint_get', __NAMESPACE__ . '\add_public_coming_soon_to_settings_endpoint_get' );
+
+/**
+ * This filter makes sure it's possible to change `wpcom_public_coming_soon` option via public-api.wordpress.com.
+ *
+ * @param object $input Filtered POST input.
+ * @param object $unfiltered_input Raw and unfiltered POST input.
+ *
+ * @return mixed
+ */
+function add_public_coming_soon_to_settings_endpoint_post( $input, $unfiltered_input ) {
+	if ( array_key_exists( 'wpcom_public_coming_soon', $unfiltered_input ) ) {
+		$input['wpcom_public_coming_soon'] = (int) $unfiltered_input['wpcom_public_coming_soon'];
+	}
+	return $input;
+}
+add_filter( 'rest_api_update_site_settings', __NAMESPACE__ . '\add_public_coming_soon_to_settings_endpoint_post', 10, 2 );
+
+/**
+ * Launch the site when the privacy mode changes from public-not-indexed
+ * This can happen due to clicking the launch button from the banner
+ * Or due to manually updating the setting in wp-admin or calypso settings page.
+ *
+ * @param  string $old_value the old value of blog_public.
+ * @param  string $value     the new value of blog_public.
+ * @return bool              whether an update occurred.
+ */
+function disable_coming_soon_on_privacy_change( $old_value, $value ) {
+	if ( 0 !== (int) $old_value || 0 === (int) $value ) {
+		// Do nothing if not moving from public-not-indexed.
+		return false;
+	}
+	update_option( 'wpcom_public_coming_soon', 0 );
+	return true;
+}
+add_action( 'update_option_blog_public', __NAMESPACE__ . '\disable_coming_soon_on_privacy_change', 10, 2 );
+
+// phpcs:disable Generic.CodeAnalysis.UnusedFunctionParameter.FoundBeforeLastUsed
```

Comment:
Looks like this isn't needed in the monorepo. The enable on line 220 could go too then.

Reply:
c7224854e276bd79277dc1894775808a307ed2f8